src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -157,6 +158,11 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
             }
         }
 
+        @Override public byte[] getOutput(FilePath workspace, Launcher launcher) throws IOException, InterruptedException {
+            // TODO could perhaps be more efficient for large files to send a MasterToSlaveFileCallable<byte[]>
+            return IOUtils.toByteArray(getOutputFile(workspace).read());
```

Comment:
If you are retrieving a file large enough to have to ask this question, you have got bigger problems, since the return value is going to be serialized in `program.dat`, etc.